mongodbatlas/data_source_mongodbatlas_cloud_shared_tier_restore_jobs.go

```diff
@@ -0,0 +1,135 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+// This datasource does not have a resource: we tested it manually
+func dataSourceMongoDBAtlasCloudSharedTierRestoreJobs() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMongoDBAtlasCloudSharedTierRestoreJobRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"results": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"job_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"target_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"target_deployment_item_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"snapshot_finished_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"restore_scheduled_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"restore_finished_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"delivery_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"expiration_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"total_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasCloudSharedTierRestoreJobRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).AtlasV2
+
+	projectID := d.Get("project_id").(string)
+	clusterName := d.Get("cluster_name").(string)
+
+	jobs, resp, err := conn.SharedTierRestoreJobsApi.ListSharedClusterBackupRestoreJobs(ctx, clusterName, projectID).Execute()
+	defer resp.Body.Close()
```

Comment:
SDK deals with the closing of the body. This is not an issue but it will result in no op. 
Was this added to make Linter happy? If yes then see:

https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_5_best_practices.md#linting-issues

Reply:
udpated